pulp_smash/tests/rpm/cli/test_upload.py

```diff
@@ -0,0 +1,105 @@
+# coding=utf-8
+"""Tests upload of units via client."""
+from __future__ import unicode_literals
+
+import os
+
+import unittest2
+
+from pulp_smash import cli, config, utils
+from pulp_smash.constants import DRPM_URL, DRPM
+from pulp_smash.tests.rpm.utils import set_up_module
+
+
+def setUpModule():  # pylint:disable=invalid-name
+    """Execute ``pulp-admin login`` on the target Pulp system."""
+    set_up_module()
+    utils.pulp_admin_login(config.get_config())
+
+
+class UploadDrpmTestCase(unittest2.TestCase):
+    """Test whether one can upload DRPMs."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create a repository and download temporary drpm."""
+        super(UploadDrpmTestCase, cls).setUpClass()
+
+        cls.cfg = config.get_config()
+        cls.repo_id = utils.uuid4()
+        cli.Client(cls.cfg).run(
+            'pulp-admin rpm repo create --repo-id {}'
+            .format(cls.repo_id).split()
+        )
+
+        completed_process = cli.Client(cls.cfg).run(
+            'mktemp --directory'.split()
+        )
+        cls.temp_folder = completed_process.stdout.strip()
+        cls.temp_drpm = os.path.join(cls.temp_folder, DRPM)
+
+        cli.Client(cls.cfg).run(
+            'curl -o {} {}'
+            .format(cls.temp_drpm, DRPM_URL).split()
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete the repository and drpm created by :meth:`setUpClass`."""
+        super(UploadDrpmTestCase, cls).tearDownClass()
+
+        cli.Client(cls.cfg).run(
+            'pulp-admin rpm repo delete --repo-id {}'
+            .format(cls.repo_id).split()
+        )
+        cli.Client(cls.cfg).run('pulp-admin orphan remove --all'.split())
+        cli.Client(cls.cfg).run('rm -rf {}'.format(cls.temp_folder).split())
+
+    def test_upload_and_output(self):
+        """Test upload of DRPM.
+
+        Steps
+
+        1. Upload DRPM
+        2. Check output of ``pulp-admin rpm repo content drpm``
+
+        This test case targets
+        `Pulp Smash #336 <https://github.com/PulpQE/pulp-smash/issues/336>`_
+        """
+        cli.Client(self.cfg).run(
+            'pulp-admin rpm repo uploads drpm --repo-id {} -f {}'
+            .format(self.repo_id, self.temp_drpm).split()
+        )
+        completed_process = cli.Client(self.cfg).run(
+            'pulp-admin rpm repo content drpm --repo-id {} --fields filename'
+            .format(self.repo_id).split()
+        )
+
+        excepted_output = (
+            '\x1b[0mFilename:     %s' % os.path.join('drpms/', DRPM)
+        )
+
+        self.assertEqual(completed_process.stdout.strip(), excepted_output)
```

Comment:
FYI, this is a more resilient approach:

```python
filename = completed_process.stdout.split('Filename:')[1].strip()
self.assertEqual(filename, 'drpms/' + DRPM)
```

The `[0m` bit of code is an ANSI color code, and is one of those things that could reasonably appear or disappear at any time.